### ISSUE TITLE
timew modify(stop, shorten, delete) error after month changed

### ISSUE THREAD

**Issue by JohannaBethge** [GothenburgBitFactory/timewarrior#360]
I started a task (timew start 'tag1' on the 31st of July, which ended only on the 1st of August. When I tried to modify the task using shorten or delete, I got an error:

```
$ timew delete @1 -> timew: /home/USERNAME/timew-1.1.1/src/Datafile.cpp:111: void Datafile::deleteInterval(const Interval&): Assertion `_range.overlap (interval.range)' failed.
Aborted (core dumped)
```

```
$timew shorten @1 60min
timew: /home/USERNAME/timew-1.1.1/src/Datafile.cpp:111: void Datafile::deleteInterval(const Interval&): Assertion `_range.overlap (interval.range)' failed.
Aborted (core dumped)

```
It might be a bug occuring, when reccording while there is a change in day and month? Maybe because a new tracking/log-file is generated for each month?

I am using the Ubuntu App on a Windows PC and timew-1.1.1

Is there a possibility to fix the issue and modify (e.g. adapt timelength or split task to multiple tasks) this tracked task?

**Comment by sruffell** [comment]
Hmm. I'm not sure if there is enough information for me to reproduce with timwarrior 1.1.1.

However, timewarrior 1.1.1 is pretty old at this point. Is it possible for you to update to 1.3.0, which is the latest release and try again?

**Comment by lauft** [comment]
I agree with @sruffell. Please try to update to 1.3.0!

Btw. I tried to reproduce your problem with the lines below
```
export TIMEWARRIORDB=/tmp/timewarriordb
rm -rf ${TIMEWARRIORDB}
mkdir -p ${TIMEWARRIORDB}/data
:> ${TIMEWARRIORDB}/timewarrior.cfg

timew start 2020-07-31T23:00 FOO
timew stop 2020-08-01T01:00
timew shorten @1 30min
timew delete @1
```
but with no error on neither `1.1.1` nor `1.3.0`, so there may something else be missing in the picture.

**Comment by JohannaBethge** [comment]
Hi, sorry, I wasn't able to upgrade to 1.3.0. If I try to download and build the 1.3.0-version. Its quite small (~3000 Byte)
`$ curl -O https://taskwarrior.org/download/timew-1.3.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2785  100  2785    0     0  12004      0 --:--:-- --:--:-- --:--:-- 12056`
 and tells me
` $ tar xzf timew-1.3.0.tar.gz

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now`

**Comment by sruffell** [comment]
Hmm :( I've not looked how the websites get updated, but it does seem l ike the taskwarrior.org site needs updating to point to the latest release.

However, the link is good on https://timewarrior.net if you want to build from a downloaded source tar ball.

```
$ curl -L -O https://github.com/GothenburgBitFactory/timewarrior/releases/download/v1.3.0/timew-1.3.0.tar.gz
$ tar xzf timew-1.3.0.tar.gz
$ cd timew-1.3.0
$ cmake -DCMAKE_BUILD_TYPE=Release .
$ make
```

**Comment by lauft** [comment]
I did some updates on the websites.

Please report missing/outdated information at the respective repository's issue trackers:
* [timewarrior.net](https://github.com/GothenburgBitFactory/ti.net/issues)
* [taskwarrior.org](https://github.com/GothenburgBitFactory/tw.org/issues)
* [gothenburgbitfactory.org](https://github.com/GothenburgBitFactory/gbf.org/issues)

Thanks

**Comment by JohannaBethge** [comment]
Hi together,
sorry for the late respond. Thank you very much @sruffell, this worked and I was able to update to version 1.3.0 (with additionally using $sudo make install).
I fixed the original problem, by chaging the data-file manually to a time before midnight. However, I realized that one problem, was that the file was still open /read by a program (might be timewarrior). So, a simple restart of the PC might have solved the problem as well (only restarting the Ubuntu-App didn't solved the problem). Since than I wasn't able to reproduce the error (e.g. by changing back to the original data-file). Hence, I guess that the error just occured, because the July-file was still read but in read-only mode (for some reasons). 
Thank you very much for your help.
